main.rb

```diff
@@ -0,0 +1,154 @@
+# rubocop:disable Metrics/PerceivedComplexity,Metrics/CyclomaticComplexity
+
+module Enumerable
+  # 1. my_each method
+
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    array = to_a
+    size.times { |index| yield array[index] }
+    self
+  end
+
+  # 2. my_each_with_index
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    arr = to_a
+    pos = 0
+    while pos < arr.length
+      yield(arr[pos], pos)
+      pos += 1
+    end
+
+    self
+  end
+
+  # 3.my_select
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    to_a.my_each { |item| new_arr << item if yield(item) }
+    new_arr
+  end
+
+  # 4.my_all?
+
+  def my_all?(args = nil)
+    if block_given?
+      counter_false = 0
+      my_each { |num| counter_false += 1 unless yield num }
+      counter_false.zero?
+    elsif args.nil
+      my_all? { |num| num }
+    else
+      my_all? { |num| args == num }
+    end
+  end
+
+  # 5.my_any?
+
+  def my_any?(args = nil)
+    if block_given?
+      counter_true = 0
+      my_each { |num| counter_true += 1 if yield num }
+      counter_true.positive?
+    elsif args.nil?
+      my_any? { |num| num }
+    else
+      my_any? { |num| args == num }
+    end
+  end
+end
+
+# 6.my_none?
+
+def my_none?(args = nil)
+  if block_given?
+    counter_true = 0
+    my_each { |num| counter_true += 1 if yield num }
+    counter_true.zero?
+  elsif args.nil?
+    my_none? { |num| num }
+  else
+    my_none? { |num| args == num }
+  end
+end
```

Comment:
my_none?:

When no block or argument is given returns true only if none of the collection members is true
`p [nil, false, true, []].my_none?` should return false

When a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
`p ['hello'].my_none?(5)`  should return true

when a class is passed as an argument returns true if none of the collection is a member of such class
`p [3,3,3,3].my_none?(String)` should return true

when a Regex is passed as an argument returns true only if none of the collection matches the Regex
`p ['hello','milk','milo'].my_none?(/z/)` should return true